router/router.go

```diff
@@ -1,40 +1,47 @@
 package router
 
 import (
-	"encoding/json"
 	"fmt"
 	"math"
 	"math/rand"
 	"sort"
-	"strconv"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
 	"github.com/rudderlabs/rudder-server/router/throttler"
 	"github.com/rudderlabs/rudder-server/services/diagnostics"
-	uuid "github.com/satori/go.uuid"
 
 	"github.com/rudderlabs/rudder-server/config"
 	"github.com/rudderlabs/rudder-server/jobsdb"
-	"github.com/rudderlabs/rudder-server/processor/integrations"
 	"github.com/rudderlabs/rudder-server/rruntime"
 	"github.com/rudderlabs/rudder-server/services/customdestinationmanager"
-	destinationdebugger "github.com/rudderlabs/rudder-server/services/destination-debugger"
 	"github.com/rudderlabs/rudder-server/services/stats"
 	"github.com/rudderlabs/rudder-server/utils/logger"
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
+type WorkerFactory interface {
+	CreateWorkers(rt *HandleT)
```

Comment:
I would probably call this LaunchWorkers rather than CreateWorkers

Reply:
Shouldn't this also take worker count? Because you are accessing workers using idx in next function. You need to know how many workers are present right?